### PR TITLE
feat:cron job for snyk token validation

### PR DESCRIPTION
**Pull request by prashbnair**

### CODE REVIEW

File: f8a_report/helpers/db_gateway.py [hunk 1]
```diff
@@ -105,3 +106,40 @@ def retrieve_ingestion_results(self, start_date, end_date) -> dict:
...
+                sql.SQL("select user_id, snyk_api_token from user_details where status=\'%s\'")
+            self.cursor.execute(get_registered_user_sql.as_string(self.conn)
+                                % UserStatus.REGISTERED.name)
+            result = self.cursor.fetchall()
+            user_id_token_dict = {row[0]: row[1] for row in result}
+            return user_id_token_dict
+        finally:
+            self.conn.close()
+
+    def update_users_to_unregistered(self, unregistered_users: list):
```
Comment (arajkumar) [inline f8a_report/helpers/db_gateway.py:131]: Should we update `updated_date` field for registered users as well?

File: tests/test_snyk_token_validation.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class TestSnykTokenValidation(unittest.TestCase):
+    """Test cases for checking Snyk Token Validation."""
+
+    def test_registered_user(self):
+        """Test case for checking registered user."""
+        with patch('f8a_report.snyk_token_validation_main.decrypt_api_token') \
+                as decrypt_api_token:
+            with patch('f8a_report.snyk_token_validation_main.is_snyk_token_valid') \
```
Comment (arajkumar) [inline tests/test_snyk_token_validation.py:15]: Usual approach is to use `patch` decorator rather than nested with statements.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: f8a_report/helpers/db_gateway.py (hunk 1 above)
Comment (prashbnair) [reply f8a_report/helpers/db_gateway.py:131]: I dont think so, as we are not updating anything for registered users and also the job will run for all registered users. It is not an interval based check

File: tests/test_snyk_token_validation.py (hunk 2 above)
Comment (prashbnair) [reply tests/test_snyk_token_validation.py:15]: Ok

File: f8a_report/helpers/db_gateway.py [hunk 3]
```diff
@@ -105,3 +106,40 @@ def retrieve_ingestion_results(self, start_date, end_date) -> dict:
...
+
+
+class TokenValidationQueries(Postgres):
+    """Snyk Token Validation Queries."""
+
+    def __init__(self):
+        """Class Constructor."""
+        super().__init__()
+
+    def get_registered_user_tokens(self) -> dict:
```
Comment (deepak1725) [inline f8a_report/helpers/db_gateway.py:118]: i feel this query should go in `v2/db_gateway.py`

**Review by deepak1725 — CHANGES_REQUESTED** [review]

**Review by deepak1725 — APPROVED** [review]

**Comment by prashbnair** [comment]
@arajkumar @yzainee Can you please take a look at this PR?

**Review by yzainee — APPROVED** [review]
LGTM. Please check one thing. For the select query, just cross verify whats the limit for the max rows it returns. If there is some limit like say (100), then you might have to add logic to read next page (pagination)

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**